### DISCUSSION TITLE
Clean Install problem through installer.sh

### DISCUSSION THREAD

**Discussion by v4lt0n** [telekom-security/tpotce#1322] [Q&A]
Hi,
I installed tpot on my debian 11 bulleye I installed it with installer type conf but it is failing without any reason and i cannot see any log only journalctl.
![image](https://user-images.githubusercontent.com/114784043/236191191-5cca37d8-8065-435e-a97f-41f9233219c8.png)

- 🔍 Use the [search function](https://github.com/dtag-dev-sec/tpotce/issues?utf8=%E2%9C%93&q=) first
- 🧐 Check our [WIKI](https://github.com/dtag-dev-sec/tpotce/wiki)
- 📚 Consult the documentation of 💻 [Debian](https://www.debian.org/doc/), 🐳 [Docker](https://docs.docker.com/), the 🦌 [ELK stack](https://www.elastic.co/guide/index.html) and the 🍯 [T-Pot Readme](https://github.com/dtag-dev-sec/tpotce/blob/master/README.md).
- **⚠️ Provide [basic support information](#info) or similiar information with regard to your issue or we can not help you and will close the issue without further notice**

<br>
<br>
<br>

<a name="info"></a>
## ⚠️ Basic support information (commands are expected to run as `root`)

- What version of the OS are you currently using `lsb_release -a` and `uname -a`?
- **Debian 11 bullleye**
- What T-Pot version are you currently using?
- **last tpot installer**
- What edition (Standard, Nextgen, etc.) of T-Pot are you running?
- **standard**
- What architecture are you running on (i.e. hardware, cloud, VM, etc.)?
- cloud kvm
- Did you have any problems during the install? If yes, please attach `/install.log` `/install.err`.
- **i cannot find install.log** 
- How long has your installation been running?
- **it finished fast but service of tpot wont run it is ain a loop**
- Did you install upgrades, packages or use the update script?
- **no**
- Did you modify any scripts or configs? If yes, please attach the changes.
- **nothing only changes pass and user in conf** 
- Please provide a screenshot of `glances` and `htop`.
- **yes please find below** 
- How much free disk space is available (`df -h`)?
-  **i have other 500gb free**
- What is the current container status (`dps.sh`)?
- **down**
- What is the status of the T-Pot service (`systemctl status tpot`)?
- **DOWN**
- What ports are being occupied? Stop T-Pot `systemctl stop tpot` and run `netstat -tulpen`
- **no open ports clean install** 
- If a single container shows as `DOWN` you can run `docker logs <container-name>` for the latest log entries
 it wont show any containers 
htop : 
![image](https://user-images.githubusercontent.com/114784043/236191821-dd51a27c-9f24-4be3-8bc5-0e3d2c7363c8.png)

**Comment by v4lt0n** [comment]
I also ran update.sh but still loophole restarting .

**Comment by Ganesh-403** [comment]
@v4lt0n 
The error:

```text
A dependency job for tpot.service failed.
```

doesn't identify the root cause by itself—it only means that **one of the prerequisites for `tpot.service` failed**, so T-Pot never reached the point of creating its containers.

From your description:

- ✅ Fresh install
- ✅ No containers created (`dps.sh` shows DOWN)
- ✅ No ports in use
- ✅ `update.sh` didn't help

this suggests the failure is happening **before Docker Compose starts the stack**, rather than being caused by an individual honeypot.

I'd recommend checking the following in order:

### 1. Inspect the systemd logs

Instead of just `journalctl -xe`, check the service-specific logs:

```bash
systemctl status tpot.service
journalctl -xeu tpot.service
```

These usually show which dependency or command failed.

---

### 2. Check whether Docker is healthy

Since T-Pot depends entirely on Docker, verify it first:

```bash
systemctl status docker
docker info
docker version
```

If Docker isn't running correctly, `tpot.service` will fail immediately.

---

### 3. Try bringing up the compose stack manually

This often produces a much more useful error than systemd:

For newer releases:

```bash
cd ~/tpotce
docker compose up
```

For older releases:

```bash
docker-compose -f /opt/tpot/etc/tpot.yml up
```

If there's a YAML error, missing image, or configuration problem, it should be printed directly.

---

### 4. Check the installation logs

Although you mentioned you couldn't find `install.log`, it should normally be generated during installation. If it isn't present, that may indicate the installer exited unexpectedly before completion.

Also check:

```bash
cat ~/tpotce/data/tpotinit.log
```

if the directory exists.

---

### 5. Verify system requirements

A clean installation failing before any containers start can also be caused by:

- Docker daemon issues
- unsupported kernel features
- storage driver problems
- cloud image restrictions
- network/DNS issues preventing image downloads

For example, there have been reports where Docker images failed to download because the host couldn't resolve Docker Hub, resulting in a seemingly successful installation but no runnable containers. :contentReference[oaicite:0]{index=0}

---

At this point, the most helpful information would be the output of:

```bash
systemctl status tpot.service
journalctl -xeu tpot.service
systemctl status docker
docker info
docker compose up
```

That should reveal whether this is:

- a Docker issue,
- a Docker Compose/configuration issue,
- an image pull failure,
- or another dependency preventing `tpot.service` from starting.

If this solves your problem, feel free to mark it as the accepted answer so others can find it easily.
